File: Dsa-day229.cpp
class Solution {
public:
    int numSpecial(vector<vector<int>>& mat) {
        int ans=0;
        for(int i=0; i<mat.size(); i++){
            for(int j=0; j<mat[0].size(); j++){
                if(mat[i][j]==1){
                    //cout<<i<<" "<<j;
                    bool b= true;
                    for(int k=0; k<mat.size(); k++){
                        if(k!=i && mat[k][j]==0) continue;
                        if(k==i) continue;
                        else {
                            b=false;
                            break;
                        }
                    }
                    if(b){
                        for(int k=0; k<mat[0].size(); k++){
                        if(k!=j && mat[i][k]==0) continue;
                        if(k==j) continue;
                        else {
                            b=false;
                            break;
                        }
                    }
                    if(b) ans++;
                    }
                }
                
            }
        }
        return ans;
    }
};

File: Dsa-day229.py
class Solution:
    def trav(self, i: int, j: int, mat: List[List[int]]) -> bool:
        specialCnt = 0
        for n in range(0, len(mat)):
            if mat[n][j] == 1:
                specialCnt+=1
            if specialCnt > 1:
                return False
        for n in range(0, len(mat[0])):
            if mat[i][n] == 1:
                specialCnt+=1
            if specialCnt > 2:
                return False
        return True

    def numSpecial(self, mat: List[List[int]]) -> int:
        totalSpecial = 0
        for i in range(0, len(mat)):
            for j in range(0, len(mat[0])):
                if mat[i][j] == 1 and self.trav(i, j, mat):
                    totalSpecial+=1
        return totalSpecial

                    